include/envoy/http/codec.h

```diff
@@ -115,8 +115,10 @@ enum class StreamResetReason {
   RemoteRefusedStreamReset,
   // If the stream was locally reset by a connection pool due to an initial connection failure.
   ConnectionFailure,
-  // If the stream was locally reset due to connection termination.
-  ConnectionTermination,
+  // If the stream was reset due to an upstream connection termination.
+  UpstreamConnectionTermination,
+  // If the stream was reset due to a downstream connection termination.
+  DownstreamConnectionTermination,
```

Comment:
Thanks for taking on the more comprehensive change suggested by @mattklein123. I have one design question here; in `codec.h`, we're largely providing an interface for codecs. We instantiate codecs in HCM for downstream and then again in the codec client for the upstream connection. Shouldn't these reset reasons be independent of concepts like upstream/downstreams?

Reply:
+1, from an initial read through I wonder if we could do this a bit more simply? Like basically just use the logic you originally had in the other PR in the HCM before the log gets written? (Obviously we should keep the stuctured logging updates which were missing from the last PR).

Reply:
Addressed it in 6014815.